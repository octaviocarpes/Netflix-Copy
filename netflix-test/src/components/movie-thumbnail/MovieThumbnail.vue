<template>
  <div v-if="!!movie" class="movie-thumbnail-container" @click="goToPlayer()">
    <img :src="movieThumbnail" alt="Movie Thumbnail" />
    <h4>{{ movieTitle }}</h4>
  </div>
</template>

<script>
import _ from "lodash";
import MetricService from "@/services/metrics.service";
export default {
  name: "MovieThubmnail",

  props: {
    movie: {
      required: true
    }
  },

  computed: {
    movieThumbnail() {
      return _.get(this.movie, "thumbnails.medium.url", "");
    },

    movieTitle() {
      return _.get(this.movie, "title", "Movie Title");
    }
  },

  methods: {
    goToPlayer() {
      MetricService.addUserMoviesMetric(this.movie.id);
      this.$router.push(`/player/${this.movie.id}`);
    }
  }
};
</script>

<style lang="scss">
@import "./style.scss";
</style>
