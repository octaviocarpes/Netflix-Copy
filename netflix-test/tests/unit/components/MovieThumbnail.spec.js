import Vue from "vue";
import MovieThumbnail from "@/components/movie-thumbnail/MovieThumbnail.vue";

describe("MovieThumbnail.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(MovieThumbnail);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});