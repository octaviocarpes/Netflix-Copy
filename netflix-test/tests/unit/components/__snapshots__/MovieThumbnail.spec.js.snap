// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieThumbnail.spec.js has the expected html structure 1`] = `<!---->`;
